src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequence.tsx

```diff
@@ -88,6 +88,21 @@ const BlastInputSequence = (props: Props) => {
     setInput(toFasta(sequence));
   }, [sequence.header, sequence.value, forceRenderCount]);
 
+  const isInputValid = input
+    ? checkSequenceValidity(input, sequenceType)
+    : true; // treat empty input as valid
+
+  const { sequences } = useBlastForm();
+  const sequenceValidityKey = index === null ? sequences.length : index;
+  const sequenceValidityContext = useContext(BlastFormContext);
+
+  useEffect(() => {
+    sequenceValidityContext?.updateSequenceValidity(
+      sequenceValidityKey,
+      isInputValid
+    );
+  }, [isInputValid]);
```

Comment:
These hook calls can be moved even further up. `useBlastForm` can definitely go just below the last `useRef` at line 71. ` useContext` right below it? And then you can move the other parts of this highlighted section to be right below them.

Reply:
whats the benefit of moving it even further? To my understanding its pretty much at the top of the code functionality where it wont be affected by any if statement unless i have missed something.

Reply:
I have moved the `useContext` at the top and the inputvalid check also but i left the useeffect where it was since it was straight after another useeffect which kind of make sense to keep them together. The code looks much neater this way, hope thats ok now.

Reply:
It is usually good to move the hooks to the top. Like Andrey mentioned in another comment, we need them at the top where it is more visible so that we don't unknowingly place any if-statements.

Reply:
@imransl i have moved it to the top where the other `useEffect` is, could you check and let me know if we need to move it higher up. If thats the case then we need to move the other `useEffect` that was already there.

Reply:
It looks better. However, the `useEffect` right below the `isInputValid` variable.

Reply:
Done